### DISCUSSION TITLE
Clarifying Nextra scope & responsibilites - what is your vision?

### DISCUSSION THREAD

**Discussion by OskarHulter** [shuding/nextra#195] [Q&A]
Hi!
Really cool project, glad I found it!

I've noticed that there is a theme etc. in the blog example. 

In my current next+mdx project I'm using Chakra UI and I'm loving it. How does the nextra developers view the responsibilites of this package - is the goal a onestop-shop "batteries included" kind of thing? customizable?

I prefer to have tools with specific responsibilities to ease composition and reduce bloat and overlap.

1. Is the current theme and styling related packages for demonstration purposes or a central part of your vision for nextra? 
2. Is it possible to remove all styling related things or are there any technical reason for its inclusion?

I'm basically wondering if this is the correct abstraction for outsourcing the next-mdx setup code that is currently littering my code or if the scope of Nextra is bigger.

**Comment by OskarHulter** [comment]
And any input on typescript support would be great as well :)

**Reply by shuding** [reply]
Good call, but I don't see how TypeScript helps for Nextra users (it does help contributors). Nextra can't be used as a lib so there's no type suggestions for API methods.

**Reply by lluiscab** [reply]
Wouldn't publishing typings to npm help users that use a custom theme? As you said on https://github.com/shuding/nextra/discussions/195#discussioncomment-979675, anyone can create their own theme (Even tho there's not much documentation).

**Comment by shuding** [comment]
Hi, Nextra creator here. Currently this project contains 3 packages, `nextra`, `nextra-theme-docs` and `nextra-theme-blog` (being developed in the core `branch`). 

The main package `nextra` is just a Next.js plugin that processes your Markdown routes and generate normalized data (load MDX valid React components, parse front matter, parse page map, etc.). It **doesn't** handle how you render those data (so no styling related things at all).

The other 2 themes are just layout components to render the data generated by `nextra` into a Next.js page (here's how you style those data).

Regarding the vision, `nextra` will remain its scope and try to support parsing more complicated structures. The two themes will keep involving but also be opinionated. They're also examples so people can also create their own themes, for example https://github.com/jaredpalmer/nextra-blank-custom-theme is a good start.

**Reply by OskarHulter** [reply]
Thank you for the detailed breakdown and tips, exactly what I was wondering. Great job on this package, I wish you good luck on your journey!

**Reply by mi-hol** [reply]
> Regarding the vision, `nextra` will remain its scope and try to support parsing more complicated structures. The two themes will keep involving but also be opinionated. They're also examples so people can also create their own themes, for example https://github.com/jaredpalmer/nextra-blank-custom-theme is a good start.

@shuding  As I had similar questions, would it make sense to put this valuable information into README.md?
